### PR TITLE
Add `BridgeExtContext.delete()` method

### LINKED ISSUE #1343: [ext.bridge] 'BridgeExtContext' object has no attribute 'delete'
### Summary

Calling `await ctx.delete()` on a message-based bridge command raises an exception.

### Reproduction Steps

• Create a bridge command
• Call `await ctx.reply(...)` followed by `await ctx.delete()`
• Run the command from a message

### Minimal Reproducible Code

```python
import discord
from discord.ext import bridge

intents = discord.Intents.default()
intents.message_content = True

bot = bridge.Bot(command_prefix=".", intents=intents)

@bot.bridge_command()
async def test_delete(ctx: bridge.BridgeContext):
    """Test ctx.delete()"""
    await ctx.reply("Message")
    await ctx.delete()

bot.run(...)
```

### Expected Results

The bot should post a message, which should then be deleted.

### Actual Results

The command works properly when invoked as a slash command; however, when invoked as a **_message command_**, the following exception is raised:

```
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.10/site-packages/discord/client.py", line 382, in _run_event
    await coro(*args, **kwargs)
  File "main.py", line 70, in on_command_error
    await bridge_error(ctx, error)
  File "main.py", line 89, in bridge_error
    raise error from error
  File "/opt/homebrew/lib/python3.10/site-packages/discord/ext/commands/core.py", line 181, in wrapped
    ret = await coro(*args, **kwargs)
  File "main.py", line 48, in test_delete
    await ctx.delete()
AttributeError: 'BridgeExtContext' object has no attribute 'delete'
```

### Intents

default + message contents

### System Information

- Python v3.10.4-final
- py-cord v2.0.0-beta
    - py-cord pkg_resources: v2.0.0b7
- aiohttp v3.8.1
- system info: Darwin 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:47:26 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T8101

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

This bug is present both in b7 and in master.

### PR DESCRIPTION
**Pull request by baronkobama**
## Summary

This PR attempts to add a `BridgeExtContext.delete()` method. The method itself only tries to delete if a response has been made and currently doesn't raise an exception if a response has not been made.

The `ApplicationContext.delete()` method now has new documentation and allows for the `delay` parameter.

Closes #1343

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Review by EmreTech — APPROVED** [review]

File: discord/ext/bridge/context.py [hunk 1]
```diff
@@ -141,13 +141,9 @@ class BridgeExtContext(BridgeContext, Context):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self._original_response_message: Optional[Message] = None
```
Comment (plun1331) [inline discord/ext/bridge/context.py:144]: Why was this moved?

File: discord/ext/commands/context.py [hunk 2]
```diff
@@ -396,4 +397,22 @@ async def send_help(self, *args: Any) -> Any:
 
     @discord.utils.copy_doc(Message.reply)
     async def reply(self, content: Optional[str] = None, **kwargs: Any) -> Message:
-        return await self.message.reply(content, **kwargs)
+        msg = await self.message.reply(content, **kwargs)
+        if self._original_response_message is None:
+            self._original_response_message = msg
+        return msg
+
+    async def delete(self, *, delay: Optional[float] = None, reason: Optional[str] = None) -> None:
```
Comment (plun1331) [inline discord/ext/commands/context.py:405]: I feel like this should be in BridgeContext, as it doesn't really make sense as a part of prefixed commands (there's no explicit response)

File: discord/ext/bridge/context.py (hunk 1 above)
Comment (EmreTech) [reply discord/ext/bridge/context.py:144]: Well this feature PR isn't just exclusively for `ext.bridge`, but it's also for `ext.commands`. So moving it would make sense.

Comment (baronkobama) [reply discord/ext/bridge/context.py:144]: Moved it to Context so it could be used for the new `Context.delete()` method.

File: discord/ext/commands/context.py (hunk 2 above)
Comment (baronkobama) [reply discord/ext/commands/context.py:405]: That can definitely be done, just figured there could be use cases wherein a user replies via a regular Context object and may want to delete said response later on.

Comment (plun1331) [reply discord/ext/commands/context.py:405]: That could be handled by the user storing the value of `reply`.

This just doesn't make much sense as there are a few different responses to a prefixed command, it could be a DM, reaction, reply, normal message, etc.

**Review by Dorukyum — CHANGES_REQUESTED** [review]
Small change for consistency

**Review by Dorukyum — APPROVED** [review]

**Review by krittick — APPROVED** [review]
